src/components/data/data.ts

```diff
@@ -1,11 +1,16 @@
 export interface Certificate {
   id?: number;
-  supplier: string;
+  supplier: Supplier;
   certificateType: string;
   validFrom: Date;
   validTo: Date;
   preview?: string;
 }
+export interface Supplier {
+  name: string;
+  s_index?: number;
```

Comment:
There is no need for "s" prefix, "index" is enough. Also pls for object properties use "camelCase" and not "camel_case"